src/tabular/encoder.ts

```diff
@@ -0,0 +1,115 @@
+import { Tensor, Tensor1D } from "@tensorflow/tfjs-core";
+import { _zeros, getDataByType } from "../utils";
+
+/**
+ * Encodes an array, Tensor or Danfo Series using unique labels
+ * @param {data} data [Array|Series]
+ * @returns Array
+ */
+class LabelEncoder {
+  public labels: Array<any>
+  private _data: Array<any>
+
+  constructor(data: Tensor1D[] | Array<any> | any){
+    if (Array.isArray(data)) {
+      this._data = data;
+    } else if (data instanceof Tensor) {
+      this._data = data.arraySync();
+    } else {
+      throw new TypeError("data must be one of Array or Tensor1D");
+    }
+  }
+
+  /**
+   * Maps data to unique integer labels
+   * @param {data} data [Array|Series|Tensor1D]
+   * @returns Array
+   */
+  public fit(): Array<any> {
+    const data_set = new Set(this._data);
+    this.labels = Array.from(data_set);
+    const _label = this.labels;
+    let encoded_data = this._data.map((x: any) => {
+      return _label.indexOf(x);
+    });
```

Comment:
This is fine for now. I do not think we could improve the performance here. Right now the complexity is O(N^2) to me.